functions/src/db/GroupDao.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    arrayRemove,
+    arrayUnion,
+    db,
+    deleteField,
+    serverTimestamp,
+} from '../firebase/firebase'
+import { COLLECTIONS } from './constants'
+import { GroupNotFoundError } from '../api/errors/HttpError'
+import { Group, GroupId, PubicGroup } from '../types/Group'
+import { UID } from '../types/uid'
+import { Member } from '../types/Member'
+
+export interface GroupEditData {
+    id: GroupId
+    name?: string
+}
+
+export interface DatabaseGroup extends Group {
+    membersDetails: Member[]
+}
+
+export class GroupDao {
```

Comment:
pls consider creating a generic `Dao` in order to factor basic CRUD operations in a single spot

Reply:
I will consider it :D